docs/docsite/rst/dev_guide/developing_modules_documenting.rst

```diff
@@ -10,6 +10,88 @@ which conforms to the schema defined below. You may find it easier to
 start writing your ``DOCUMENTATION`` string in an editor with YAML
 syntax highlighting before you include it in your Python file.
 
+All modules must have the following sections defined in this order:
+
+1. ANSIBLE_METADATA
+2. DOCUMENTATION
+3. EXAMPLES
+4. RETURNS
+5. Python imports
```

Comment:
I'd make these links to the relevant sections below.

For linking in general we need to think about how we want to use ref: http://www.sphinx-doc.org/en/stable/markup/inline.html#ref-role  Currently our ref targets are all thrown into the same namespace.  The documentation has gotten big enough that we should think of splitting that a bit.  Probably having a set of high level categories and  prefixing the category name to the ref makes sense.  (That way we can have multiple references to an idea like:  cfg_pipelining, plugin_connection_pipelining, intro_pipelining)  Maybe dev_guide is a sufficient prefix or perhaps dev_mods?